src/fmu/dataio/_export_item.py

```diff
@@ -318,11 +366,66 @@ def _data_process_object(self):
 
         if self.subtype == "RegularSurface":
             self._data_process_object_regularsurface()
+        elif self.subtype == "Grid":
```

Comment:
Should this be more specific? E.g. `CornerPointGrid`? This is hard-coded further down, but what happens if someone tries to use this with non-cornerpoint grid?

Also, is `Grid` different from `GridProperty` or is the latter an attribute of the former? Ref offline discussions.